propagation.go

```diff
@@ -0,0 +1,100 @@
+package opentracing
+
+///////////////////////////////////////////////////////////////////////////////
+// CORE PROPAGATION INTERFACES:
+///////////////////////////////////////////////////////////////////////////////
+
+// Injector is responsible for injecting Span instances in a manner suitable
+// for propagation via a format-specific "carrier" object. Typically the
+// injection will take place across an RPC boundary, but message queues and
+// other IPC mechanisms are also reasonable places to use a Injector.
+//
+// See Extractor and Span.Injector.
+type Injector interface {
+	InjectSpan(span Span, carrier interface{}) error
+}
+
+// Extractor is responsible for extracting Span instances from an
+// format-specific "carrier" object. Typically the extraction will take place
+// on the server side of an RPC boundary, but message queues and other IPC
+// mechanisms are also reasonable places to use a Extractor.
+//
+// See Injector and Tracer.Extractor.
+type Extractor interface {
+	JoinTrace(operationName string, carrier interface{}) (Span, error)
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// BUILTIN PROPAGATION FORMATS:
+///////////////////////////////////////////////////////////////////////////////
+
+// BuiltinFormat is the shared type for builtin OpenTracing in-band
+// propagation formats.
+type BuiltinFormat int
+
+const (
```

Comment:
@bensigelman is it worth adding a comment about the closeness of this enum? I.e. that the API does not prevent extensions, because the factory method's actual argument is `interface{}`